src/api/campaigns/controller.ts

```diff
@@ -53,7 +57,13 @@ export const createCampaign = async (body: any, next: NextFunction) => {
         }
       });
     } else {
-      await sendMail(mailList.emails, newCampaign.subject, Body, newCampaign.senderMail);
+      let mailArray = [];
+      while (mailList.emails.length > 0) {
+        mailArray.push(
+          sendMail(mailList.emails.splice(0, emailBatchSize + 1), newCampaign.subject, Body, newCampaign.senderMail),
+        );
+      }
+      await Promise.all(mailArray);

```

Comment:
Promise.all will work here if all the mails are successfully delivered without any bounces. What if one of the promises is rejected but the rest are resolved? In that case, the API will throw an error despite sending the mails. Instead of promise.all, you can use promise.allSettled which waits for all promises to be settled. 

Refer the mdn docs for more details. 

1. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all
2. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

@Developer-R-7 @Srivastava57Harsh You need to come up with a strategy to make sure the mails sent and mails not sent are segregated and the user has a way of seeing them. For example, if the first 49 mails are sent, but the next 49 are not sent, send a message to user saying, we were unable to send mails to some people. Now, you can either list those mails, or give a unique id or a link and then show those mails. For that, you have to store the rejected mails in a db. 

I leave it up to you to come up with a solution.